rhizome/host/bin/prep_host.rb

```diff
@@ -4,6 +4,13 @@
 require_relative "../../common/lib/util"
 require_relative "../lib/cloud_hypervisor"
 require "fileutils"
+require "socket"
+
+original_hostname = Socket.gethostname
+hostname = ARGV.shift
+
+File.write("/etc/hosts", File.read("/etc/hosts").gsub(original_hostname, hostname))
```

Comment:
It is not safe to write to a file directly. If operation crashes midway, it can leave the file in corrupted state. Also not applicable for this case, but there can be concurrent modifications. We have "safe_write_to_file" utility in rhizome/common/lib/util.rb. It writes to temp file and do atomic rename to handle first problem and locking for the second. You can use that instead.